### PR TITLE
let plank clean up stale pending jobs

### PR DESCRIPTION
**Pull request by krzyzacy**
plank version of https://github.com/kubernetes/test-infra/pull/6883

/assign @BenTheElder @cjwagner

### CODE REVIEW

**Comment by BenTheElder** [comment]
> This LGTM except that it will delete pods for any type of job and I thought we only wanted to delete pods for periodic jobs in order to avoid killing presubmits that are waiting to be scheduled?

I think this is OK if we bump the time a bit with the intention of reducing it later once #6865 is a bit further along.

**Comment by BenTheElder** [comment]
[test comment pls ignore]
/retest

**Comment by BenTheElder** [comment]
[test comment pls ignore]
/test pull-test-infra-bazel

**Comment by krzyzacy** [comment]
/hold
after a second look - sinker does not clean pending pods at all, I'll make plank also delete the stale pod.

**Comment by krzyzacy** [comment]
/hold cancel
bumped abort timeout to 60m
and sinker should only clean pending pods if their prowjob is completed (aborted)

We want to kill stale presubmit pods and update back to github, and people can notice the issue and ping us so we can fix their job, rather than let a presubmit sitting for days and stays pending without any feedback though. And people knows how to `/retest`

File: prow/cmd/sinker/main.go [hunk 1]
```diff
@@ -186,7 +186,9 @@ func (c *controller) clean() {
 				// deleting the pod now will result in plank creating a brand new pod
 				continue
 			}
-			if (pod.Status.Phase == kube.PodSucceeded || pod.Status.Phase == kube.PodFailed) &&
+			if (pod.Status.Phase == kube.PodSucceeded ||
```
Comment (kargakis) [inline prow/cmd/sinker/main.go]: Just remove these checks? https://github.com/kubernetes/test-infra/pull/6883#issuecomment-367704769

Comment (krzyzacy) [reply prow/cmd/sinker/main.go]: done

**Pull request merged**